uint8 channel
int16 rotation